lib/ansible/modules/cloud/amazon/aws_sgw_facts.py

```diff
@@ -0,0 +1,354 @@
+    """
+    def list_gateway_file_shares(self, gateway):
+        try:
+            response = self.client.list_file_shares(
+                GatewayARN=gateway["gateway_arn"],
+                Limit=100
+            )
+
+            gateway["file_shares"] = []
+            marker = self._read_gateway_fileshare_response(gateway["file_shares"], response)
+
+            while marker is not None:
+                response = self.client.list_file_shares(
+                    GatewayARN=gateway["gateway_arn"],
+                    Marker=marker,
+                    Limit=100
+                )
+
+                marker = self._read_gateway_fileshare_response(gateway["file_shares"], response)
+        except (BotoCoreError, ClientError) as e:
+            self.module.fail_json_aws(e, msg="Couldn't list gateway file shares")
+
+    """
+    List storage gateway local disks
+    """
+    def list_local_disks(self, gateway):
+        try:
+            gateway['local_disks'] = [camel_dict_to_snake_dict(disk) for disk in
+                                      self.client.list_local_disks(GatewayARN=gateway["gateway_arn"])['Disks']]
+        except (BotoCoreError, ClientError) as e:
+            self.module.fail_json_aws(e, msg="Couldn't list storage gateway local disks")
+
+    """
+    Read tape objects from AWS API response.
+    Drop the gateway_arn attribute from response, as it will be duplicate with parent object.
+    """
+    @staticmethod
+    def _read_gateway_tape_response(tapes, aws_response):
+        for tape in aws_response["TapeInfos"]:
+            tape_obj = camel_dict_to_snake_dict(tape)
+            if "gateway_arn" in tape_obj:
+                del tape_obj["gateway_arn"]
+            tapes.append(tape_obj)
+
+        return aws_response["Marker"] if "Marker" in aws_response else None
+
+    """
+    List VTL & VTS attached to AWS storage gateway in VTL mode
+    """
+    def list_gateway_vtl(self, gateway):
+        try:
+            response = self.client.list_tapes(
+                Limit=100
+            )
+
+            gateway["tapes"] = []
+            marker = self._read_gateway_tape_response(gateway["tapes"], response)
+
+            while marker is not None:
+                response = self.client.list_tapes(
+                    Marker=marker,
+                    Limit=100
+                )
+
+                marker = self._read_gateway_tape_response(gateway["tapes"], response)
+        except (BotoCoreError, ClientError) as e:
+            self.module.fail_json_aws(e, msg="Couldn't list storage gateway tapes")
+
+    """
+    List volumes attached to AWS storage gateway in CACHED or STORAGE mode
+    """
+    def list_gateway_volumes(self, gateway):
+        try:
+            paginator = self.client.get_paginator('list_volumes')
+            response = paginator.paginate(
+                GatewayARN=gateway["gateway_arn"],
+                PaginationConfig={
+                    'PageSize': 100,
```

Comment:
Not sure the PaginationConfig is necessary - we don't have it in any other modules.